Makefile

```diff
@@ -67,6 +67,11 @@ endif
 tesh:
 	@poetry run tesh *.md
 
+.PHONY: examples
+examples:
+	@poetry run tesh examples/happy.md
+	@poetry run tesh examples/nix-shell.md
```

Comment:
Some thoughts:

1. I know this is a bit an Inception-style thinking, but happy.md is in fact already tested:
* `@poetry run tesh *.md` runs tesh on `README.md`
* `README.md` has a `console` section that runs `examples/happy.md`
* it probably does not make sense to run `examples/happy.md` twice

2. We should probably run "poetry run tesh examples/" so that if anyone adds another example, it is automatically included in CI runs. But then the `console` section in `README` will grow too large. Also, debugging it could be not-fun as there are two levels of indirection.


## Suggestion

1. Leave `demo/` as is, run it in `console` section in `README.md`
2. Add `examples/` folder, drop `nix-shell.md` in there.
3. Run `poetry run tesh examples/` in `make examples`. 

4.